secondarylib/src/main/scala/com/socrata/datacoordinator/secondary/SecondaryWatcher.scala

```diff
@@ -314,26 +315,32 @@ object SecondaryWatcherClaimManager {
   private val log = LoggerFactory.getLogger(classOf[SecondaryWatcherClaimManager])
 
   // in-memory list of datasets that are actively being worked on by _this_ instance
-  private val workingSet = new mutable.HashSet[(String, Long)] with mutable.SynchronizedSet[(String, Long)]
+  private object X // like Unit, but nullable because Java
+  private val workingSet = new ConcurrentHashMap[(String, Long), X.type]
 
   def workingOn(storeId: String, datasetId: DatasetId): Unit = {
-    if (workingSet.contains((storeId, datasetId.underlying))) {
+    if(workingSet.putIfAbsent((storeId, datasetId.underlying), X) ne null) {
       log.error("We have already claimed dataset {} for store {} in our working set." +
         s" An unexpected error has occurred; exiting.", datasetId, storeId)
       sys.exit(1)
     }
 
-    workingSet += ((storeId, datasetId.underlying))
     log.debug(s"Added dataset {} for store $storeId to our working set which is now {}.", datasetId, workingSet)
   }
 
   def doneWorkingOn(storeId: String, datasetId: DatasetId): Unit = {
-    workingSet -= ((storeId, datasetId.underlying))
+    workingSet.remove((storeId, datasetId.underlying))
     log.debug(s"Removed dataset {} for store $storeId from our working set which is now {}.", datasetId, workingSet)
   }
 
-  def andInWorkingSetSQL: String = workingSet.headOption.map { _ =>
-    workingSet.map(_._2).mkString(" AND dataset_system_id IN (", ",", ")") }.getOrElse("")
```

Comment:
workingSet is no longer a Scala set at all. It's a Java map, because the `SynchronizedSet` trait is deprecated, as it's basically impossible to use correctly -- indeed there was a race condition in this very class that I fixed, up in `workingOn`.